pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -169,6 +214,11 @@ def detach_data(model: "pl.LightningModule") -> None:
             if isinstance(loader, _PatchDataLoader):
                 loader.unpatch(model)
 
+    def teardown(self) -> None:
+        for attr in vars(self).values():
+            if isinstance(attr, AbstractDataFetcher):
+                attr.teardown()
+
```

Comment:
could this happen in the loop teardown instead of here?

Reply:
Any reason why you think it would be better within loop teardown ? The loop don't have a reference to the data_fetchers.